EP-15 공용 인풋 컴포넌트 구현

rak517 opened the pull request:
## ❓이슈
- close #9 

## :writing_hand: Description

<!-- 어떤 내용의 PR인지 작성해주세요. (ex. 메인 페이지 레이아웃 작업) -->
<!-- ⚠️ PR에는 해당 PR의 제목에 해당하는 내용만 들어가 있어야 합니다!  -->
공용 인풋 컴포넌트에 관한 PR입니다.

## 작업 내용
1. BaseItem, BaseLabel, BaseError 등 공통 UI 요소 컴포넌트 생성
2. cva를 활용한 baseFieldClassName, baseErrorClassName 정의
3. label과 error prop을 통한 동적 에러 및 레이블 처리 지원하는 Input 컴포넌트 구현
4. BaseField 타입 정의를 위한 types.ts 파일 추가
5. Input 컴포넌트 스토리 작성

## Storybook 캡쳐
![스크린샷 2025-03-10 22 09 18](https://github.com/user-attachments/assets/a2ab399a-ba4f-461b-a73b-2335a084dea4)


## :white_check_mark: Checklist

### PR

<!-- 작성중인 PR인 경우, Draft 모드로 생성해주세요. -->

- [x] Branch Convention 확인
  > `feat/*` 기능 구현, `fix/*` 버그 수정, `refactor/*` 개선
- [x] Base Branch 확인
- [x] 적절한 Label 지정
- [x] Assignee 및 Reviewer 지정

### Test

- [x] 로컬 작동 확인

### Additional Notes

<!-- 추가 사항이 있을 경우, Todo list를 작성해주세요. -->

- [x] (없음)

rak517 committed:
:fire: remove: 빈 폴더 추적을 위한 .gitkeep 파일 삭제

rak517 committed:
:sparkles: feat: 공용 Input 및 Base 컴포넌트 구현

- BaseItem, BaseLabel, BaseError 등 공통 UI 요소 컴포넌트 생성
- cva를 활용한 baseFieldClassName, baseErrorClassName 정의
- label과 error prop을 통한 동적 에러 및 레이블 처리 지원하는 Input 컴포넌트 구현
- BaseField 타입 정의를 위한 types.ts 파일 추가

rak517 committed:
:fire: remove: 빈 폴더 추적을 위한 .gitkeep 파일 삭제

rak517 committed:
🚚 refactor: 폴더 구조 변경 및 파일 재배치

- 기존 컴포넌트들을 Field 폴더로 이동
- 관련 import 경로 수정

rak517 committed:
:sparkles: feat: Input 컴포넌트 스토리 작성

rak517 committed:
:recycle: refactor: BaseLabel cn 함수 추가 및 Input 스토리 배경 개선

CJewon commented:
코드쪽에서 문제점은 없었고 storybook에서도 확인완료했습니다 !

ToKyun02 commented on src/components/ui/Field/Input.tsx:
```diff
@@ -0,0 +1,28 @@
+  InputHTMLAttributes<HTMLInputElement> &
+  VariantProps<typeof baseFieldClassName> & {
+    ref?: Ref<HTMLInputElement>;
+    variant?: 'default' | 'outlined';
```
variant 부분은 이미 `VariantProps<typeof baseFieldClassName>`에서 정의하고 있어서 없어도 될 것 같습니다!

ToKyun02 commented on src/components/ui/Field/Base.tsx:
```diff
@@ -0,0 +1,33 @@
+}
+
+export const baseFieldClassName = cva(
+  'rounded-lg w-full px-4 h-11 lg:h-[64px] md:w-[384px] lg:w-[640px] text-black-950 placeholder-blue-400 text-lg lg:text-xl focus-visible:outline-none read-only:text-blue-400',
```
width를 반응형으로 짠 이유가 혹시 있을까요? input의 너비는 웬만해선 w-full인 것 같아서 여쭤봤습니다!

ToKyun02 commented on src/components/ui/Field/index.tsx:
```diff
@@ -0,0 +1 @@
+export * from './Input';
```
import barrel 쓰신 것 좋습니다!👍

ToKyun02 commented on src/components/ui/Field/Base.tsx:
```diff
@@ -0,0 +1,33 @@
+    },
+  },
+);
+export const baseErrorClassName = 'border border-error';
```
상수로 네이밍 하는 게 좋을 것 같습니다!

ToKyun02 commented on src/components/ui/Field/Base.tsx:
```diff
@@ -0,0 +1,33 @@
+  return <div className='grid gap-1'>{children}</div>;
+}
+
+export function BaseLabel({ required, children, className, ...props }: PropsWithChildren<LabelHTMLAttributes<HTMLLabelElement> & { required?: boolean; className?: string }>) {
```
이미 먼저 타입정의에 LabelHTMLAttributes로 className을 정의해서 뒤에 나오는 인터섹션 타입에서 className부분은 없어도 될 것 같습니다.

ToKyun02 reviewed (approved):
작업하시느라 고생하셨습니다!
리팩토링 해야하는 것들은 코멘트로 남겼습니다.

추가로, 지금 이슈 PR연결 안되어있는데, 이러면 PR닫혀도 이슈가 안닫히니 참고해주세요.

rak517 committed:
:recycle: refactor: 리뷰 사항 반영

- baseErrorClassName 네이밍 상수로 변경
- InputProps variant 유니온 타입 제거
- BaseLabel className 인터섹션 타입 제거

rak517 committed:
:recycle: refactor: Input 반응형 처리 제거
